smartapps/curb/curb-energy-manager.src/curb-energy-manager.groovy

```diff
@@ -0,0 +1,248 @@
+ *
+ */
+
+definition(name: "CURB Energy Manager", namespace: "curb", author: "Neil Zumwalde", description: "Maximize your energy savings with CURB!", category: "", iconUrl: "http://energycurb.com/wp-content/uploads/2015/12/curb-web-logo.png", iconX2Url: "http://energycurb.com/wp-content/uploads/2015/12/curb-web-logo.png", iconX3Url: "http://energycurb.com/wp-content/uploads/2015/12/curb-web-logo.png")
+
+preferences {
+  page(name: "pageOne", nextPage: "pageTwo") {
+    section("Program") {
+      input("name", "text", title: "Program Name", defaultValue: "CURB Energy Manager")
+      input("enabled", "bool", title: "Active", defaultValue: true)
+    }
+    section("When to run") {
+      input("weekdays", "enum", title: "Set Days of Week", multiple: true, required: true,
+              options: ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday", "Sunday"],
+                defaultValue: ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday"])
+      input("hours", "enum", title: "Select Times of Day", multiple: true, required: true,
+              options: [[0 : "12am"], [1 : "1am"], [2 : "2am"], [3 : "3am"], [4 : "4am"], [5 : "5am"], [6 : "6am"], [7 : "7am"], [8 : "8am"], [9 : "9am"],
+                [10 : "10am"], [11 : "11am"], [12 : "12pm"], [13 : "1pm"], [14 : "2pm"], [15 : "3pm"], [16 : "4pm"], [17 : "5pm"], [18 : "6pm"], [19 : "7pm"],
+                [20 : "8pm"], [21 : "9pm"], [22 : "10pm"], [23 : "11pm"]],
+                defaultValue: [15, 16, 17, 18, 19])
+    }
+  }
+  page(name: "pageTwo", nextPage: "pageThree" ) {
+    section("Threshold Settings") {
+      input("timeInterval", "enum", title: "Select Measurement Interval", multiple: false,
+              options: [[15 : "15 minutes"], [30 : "30 minutes"], [60 : "60 minutes"]], defaultValue: 30)
+      input("kwhThreshold", "float", title: "Set Threshold Usage (kWh)")
+            input("safetyMargin", "float", title: "Set Safety Margin (%)", defaultValue: 25)
+      input("projectionPeriod", "float", title: "Set Projection Period (%)", defaultValue: 50)
+      input("meter", "capability.energyMeter", title: "Select Monitored Circuit", multiple: false)
+      input("circuits", "capability.energyMeter", title: "Circuits to send alerts on", multiple:true, defaultValue: ALL)
+    }
+  }
+    page(name: "pageThree", install: true, uninstall:true) {
+    section("Controlled Appliances") {
+      input("thermostats", "capability.thermostat", title: "Select your Thermostat", multiple: true, required: false)
+      input("switches", "capability.switch", title: "Select your Load Controllers", multiple: true, required: false)
+            //input("cycleTimeLatency", "float", title: "Set Cycle Time Latency (minutes)", defaultValue: 5)
+    }
+
+    section("Notifications"){
+          input("recipients", "contact", title: "Send notifications to") {
+              input "phone", "phone", title: "Phone Number",
```

Comment:
this needs to be removed as it has legal implications that we currently don't comply with.

Reply:
so we can't do any notifications?

Reply:
I guess a better question would be, what do we have to do to be in compliance?

Reply:
remove any kind of sms notifications.

Reply:
Also `sendNotificationToContacts` does not work without the contact book which hasn't been deployed.
Instead use sendPush() or sendPushMessage()
http://docs.smartthings.com/en/latest/ref-docs/smartapp-ref.html?highlight=sendNotificationToContacts#sendpush